docs/semantic-search.ipynb

```diff
@@ -51,337 +58,235 @@
-            "  - I am an 18 year old college student. Is it a viable idea to play poker in order to pay for my college tuition?\n",
-            "  - If the French monarchy had never been abolished, who would be the current king/queen?\n",
-            "  - Who was the best French King?\n",
-            "  - How do I obtain a free United States phone number using the Internet?\n",
-            "  - What is the change in your opinion about PM Narendra Modi after demonetization of 1000 and 500 rupees currency notes?\n"
-          ]
-        }
-      ],
       "source": [
-        "print(\"Here are some example questions in the data set:\\n\")\n",
-        "for r in dataset.documents.iloc[0:10].to_dict(orient=\"records\"):\n",
-        "    print(\"  -\" + r['metadata']['text'])"
+        "## Creating our dataset"
       ]
     },
     {
       "cell_type": "markdown",
-      "metadata": {
-        "id": "ebd7XSamfMsC"
-      },
+      "metadata": {},
       "source": [
-        "## Creating an Index\n",
+        "We're working with a small subset of a large multilingual dataset called Tatoeba. Tatoeba consists of hundreds of thousands of sentence translation pairs, and sometimes serves as a benchmark for crosslingual semantic search capabilities.\n",
+        "\n",
+        "In this notebook, we're just testing semantic search, so we'll grab a subset of english sentences that include the word \"park\".\n",
         "\n",
-        "Now the data is ready, we can set up our index to store it.\n",
+        "Why \"park\"? In English, park has multiple meanings which occur in different contexts. It could mean a place, such as a public park. Or, it could mean an actoin with a car (to park) or a place (park-ing spot). Semantic search using embedding models will naturally distinguise between these contexts, without invervention or lablel from our selves!\n",
         "\n",
-        "We begin by instantiating the Pinecone client. To do this we need a [free API key](https://app.pinecone.io)."
+        "This is the key benefit for semantic search; a way to abstract and represent the meaning of user queries without any additional work. \n",
+        "\n",
+        "And, since our embedding model is inherently multilingual, we can even do this semantic search across several languages without any additional work!\n",
+        "\n"
       ]
     },
     {
       "cell_type": "code",
-      "execution_count": 6,
+      "execution_count": 5,
       "metadata": {},
       "outputs": [],
       "source": [
-        "import os\n",
-        "\n",
-        "if not os.environ.get(\"PINECONE_API_KEY\"):\n",
-        "    from pinecone_notebooks.colab import Authenticate\n",
-        "    Authenticate()"
+        "from datasets import load_dataset\n",
+        "# specify that we want the english-spanish translation pairs\n",
+        "tatoeba = load_dataset(\"Helsinki-NLP/tatoeba\", lang1=\"en\", lang2=\"es\", trust_remote_code=True, split=\"train\")"
```

Comment:
Just FYI, I got a warning here about HF_TOKEN. May change in the future.

<img width="1354" alt="image" src="https://github.com/user-attachments/assets/f91f0794-659b-489c-818e-04ca3b357c36" />